### DISCUSSION TITLE
Can we update TypeScript types to use ReadonlyArray when possible?

### DISCUSSION THREAD

**Discussion by ianduvall** [clauderic/dnd-kit#1034] [General]
It would be nice to update types to accept `ReadonlyArray`s otherwise consumers have to use mutable arrays or ignore flow errors. For example, [`arrayMove`](https://github.com/clauderic/dnd-kit/blob/master/packages/sortable/src/utilities/arrayMove.ts) should accept `ReadonlyArray<T>`.

**Comment by ianduvall** [comment]
@clauderic thoughts? I wouldn't mind opening a PR if you think it's a good idea.

**Reply by clauderic** [reply]
Would happily accept a PR to that effect 👍

**Reply by ianduvall** [reply]
https://github.com/clauderic/dnd-kit/pull/1040
